### DISCUSSION TITLE
Holes patterning in niobium

### DISCUSSION THREAD

**Discussion by Diego-planckian1** [iqm-finland/KQCircuits#86] [Q&A]
HI,

I need to implement holes patterning for a niobium chip? Is there anyone that has already encountered this problem?

My idea was to create a region full of boxes and subtract a Path that retrace the waveguide.

I have some problem in the usage of pya.CellInstArray() within the KQCircuits environment for the matrix of boxes....

Any help?

**Comment by qpavsmi** [comment]
This sounds similar to what we do with ground grids. In the chips of our library we define a layer called `*_ground_grid_avoidance` layer around components and waveguides, then you can generate the ground grid that covers the entire chip except for that layer. You can try it out by drag-and-dropping some chip, double-clicking it, choosing "PCell parameters" and checking the "[with_grid] Make ground plane grid". You should see the grid now when you make the `1t1_ground_grid` layer visible in `1t1-face`.

If this does not resolve your issue off the bat but you are curious to see how we implmented it, check `chip.py` function `produce_ground_on_face_grid` and `kqcircuits/util/groundgrid.py` code. We also have a similar mechanism for placing indium bumps for flipchips, check `_produce_ground_bumps` function from `chip.py`.

**Reply by Diego-planckian1** [reply]
It works! thank you very much.

I notice that the elements that I insert by my code are not avoided by the ground grid... what am I missing in my elements code?

**Reply by qpavsmi** [reply]
You need to insert some shapes into the `1t1_ground_grid_avoidance layer`. Create a shape as a `pya.Region` object and then call `add_protection` to add it. You can see how we do it in `qubits/double_pads.py` as example

**Reply by Diego-planckian1** [reply]
Okay, thank you again
